inst/rmarkdown/templates/gigascience_article/skeleton/skeleton.Rmd

```diff
@@ -0,0 +1,548 @@
+
+# Using rmarkdown features with this template
+
+Reference (among others): [https://www.rstudio.com/wp-content/uploads/2015/02/rmarkdown-cheatsheet.pdf](https://www.rstudio.com/wp-content/uploads/2015/02/rmarkdown-cheatsheet.pdf)
+
+## Sectional Headings
+
+You can use `# Section Name` and `## Subsection name` lines to add more sections and subsections to your manuscript. Further sectional levels are provided by `### Subsubsection name`, `#### Paragraph name`, `##### Subparagraph name` lines; e.g.:
+
+```
+  # Header 1
+  section
+  
+  ## Header 2
+  subsection
+  
+  ### Header 3
+  subsubsection
+  
+  #### Header 4
+  paragraph
+  
+  ##### Header 5
+  subparagraph
+  
+  ###### Header 6
+  subsubparagraph
+```
+###
+becomes:
+
+# Header 1
+section
+
+## Header 2
+subsection
+
+### Header 3
+subsubsection
+
+#### Header 4
+paragraph
+
+##### Header 5
+subparagraph
+
+###### Header 6
+subsubparagraph
+
+## Blockquotes
+
+> This is a quote. Lorem ipsum dolor sit amet, 
+> consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore 
+> magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
+> laboris nisi ut aliquip ex ea commodo consequat. 
+
+## Unordered list (bullets)
+
+- This is a point in a bullet list.
+    - A sub-point must be indented four more spaces than the parent.
+        - The same is true for a sub-sub-point.
+- A third point.
+
+## Ordered list (numbered list)
+
+1. This is a point in a numbered list.
+    - A sub-point must be indented four more spaces than the parent.
+        - The same is true for a sub-sub-point.
+1. A third point.
+
+Use R to generate tables rather than attempting to specify them in rmarkdown.
+
+## Equations
+
+```
+inline equation: $A = \pi*r^{2}$
+```
+inline equation: $A = \pi*r^{2}$
+
+```
+full-line equation: $$A = \pi*r^{2}$$
+```
+$$A = \pi*r^{2}$$
+
+## Tables
+
+### Making tables with rmarkdown
+
+Use R to generate your rmarkdown tables, e.g., see Table \ref{tab:markdown}.  Granted, literal rendering of rmarkdown tables is not working, but even if it worked, it would makes no sense to use it when you can use R to lay out tables more coherently.
+
+\begin{table}
+\caption{Table generated using rmarkdown}
+\label{tab:markdown}
+```{r setup, echo=FALSE, warning=FALSE}
+library(knitr)
+knitr::opts_chunk$set(echo = TRUE)
+
+kable(
+  t(
+    data.frame(
+      header=c("Right","Left","Center")
+    , value=c(1,2,3) 
+    )
+  )
+, align = c("r","l","c")
+, format = "latex")
+```
+\end{table}
+
+### Laying out tables in \LaTeX{}
+
+Table layout in \LaTeX{} is much less intuitive than using R, but this might be your only choice unless you define a section that has only one column per page.
+
+
+
+## Pre-existing figure
+
+It is inadvisable to generate images from R code chunks, but it's fine to include pre-existing images.
```

Comment:
The example figure below could be generated from an R code chunk, and I believe it will be more elegant and easier to maintain, too.

Reply:
The purpose of this entry is to demonstrate to the user of the template how to incorporate a pre-existing figure not generated from Rmarkdown.  So, no, this should not be changed.

Reply:
Okay. Two issues:

1. You said "It is inadvisable to generate images from R code chunks". Why? If an image is generated from R anyway, why not generate it from an R code chunk but manually generate it elsewhere and use raw LaTeX code to include it? That seems to be inconvenient to me.

2. Even with pre-existing figures, you could use `knitr::include_graphics()` to include them in an R code chunk.

In either case, I think the source code is a lot cleaner and easier to maintain when you use an R code chunk instead of raw LaTeX (see my example after four lines below).

Reply:
This template is designed to get an author (me, in particular) productive as quickly as possible for writing a journal article.  It's a port of the instructions to authors, not a "starter document".  That being said, it does need to represent best-practices.

I agree that an unsubstantiated assertion ("It is inadvisable...") is pays little regard to the needs of the reader, to say the least; I will certainly reword it to explain my perspective or remove it.

Rarely if ever would I generate graphics while generating a manuscript, because I need to generate again and again to get the text formatting perfected, and I don't want the figures potentially changing on me.   I may be generating from git clones in different environments, some of which may not have all of the R packages needed to generate the graphics for the manuscript.

I am porting the instructions-to-authors to rticles, learning as I go along.  So, thank you for pointing me to `knitr::include_graphics()` - that definitely seems to be the right tool for the job.

Reply:
Understood.